### PR TITLE
Generate code coverage reports action (iOS)

### PR DESCRIPTION
**Pull request by hadiidbouk**
Each project can have a different place to store these generated reports, we can push them to the repo using the default branch, but some projects don't allow direct push to the default branch, and instead, we should open PRs.

Also, some projects may need to store these reports outside the source code.

So it depends. This action will generate an HTML file for the code coverage and returns the generated file path.

Screenshot 2023-06-01 at 2 56 53 PM

UI updates:

Screenshot 2023-06-02 at 12 18 05 PM

### CODE REVIEW

**Review by harishsaini — APPROVED** [review]
looks good 👍

**Review by vinaykharb — APPROVED** [review]
Looks 💯

File: lib/fastlane/plugin/fueled/helper/fueled_helper.rb [hunk 1]
```diff
@@ -155,6 +155,190 @@ def self.default_gh_release_output_files
...
+        code_coverage_config_file = File.read(code_coverage_config_file_path)
+
+        if !code_coverage_config = JSON.parse(code_coverage_config_file) rescue nil
+            UI.user_error!('The provided Fueled code coverage config file isn\'t in a valid format')
+        end 
+
+        UI.message("Calculating Code Coverage Percentage... 🤓")
+        test_results_json_string = `xcrun xccov view --report --json \"#{result_bundle_file_path}\"`
+        test_results = JSON.parse(test_results_json_string)
+
```
Comment (sinchan-fueled) [inline lib/fastlane/plugin/fueled/helper/fueled_helper.rb:184]: This seems like a shell command, although it's being parsed. I got a little confused here. @hadiidbouk Can you explain this for me?

**Review by sinchan-fueled — APPROVED** [review]
Looks good to me @hadiidbouk 💪🏽 Approving!
Just had one question for my understanding.

File: lib/fastlane/plugin/fueled/helper/fueled_helper.rb (hunk 1 above)
Comment (hadiidbouk) [reply lib/fastlane/plugin/fueled/helper/fueled_helper.rb:184]: @sinchan-fueled Sure! 

You can see that we add  `--json`  to the command, which will convert the output of the command into a JSON format; we are using `JSON.parse` here to convert the JSON string into a JSON object.

We need the JSON output of the `xccov` command for easy access to the data; here, we want to see what files have been covered when running our tests.

Comment (hadiidbouk) [reply lib/fastlane/plugin/fueled/helper/fueled_helper.rb:184]: Ohh, I forgot to mention a backtick will execute the command, and it returns the output :)

**Review by thapli9A — APPROVED** [review]

File: lib/fastlane/plugin/fueled/helper/fueled_helper.rb (hunk 1 above)
Comment (sinchan-fueled) [reply lib/fastlane/plugin/fueled/helper/fueled_helper.rb:184]: Gotchu! The backtick comment cleared it for me! Thanks Hadi 😄

**Comment by metatheoretic** [comment]
@hadiidbouk can we rename `master` to `main` for this repo?

**Comment by hadiidbouk** [comment]
> @hadiidbouk can we rename `master` to `main` for this repo?

Let us warn the team about this because, as I recall, some projects use the `main` branch instead of specifying the version

**Pull request merged**